bitnami/harbor/templates/_helpers.tpl

```diff
@@ -725,131 +465,35 @@ harbor: External PostgreSQL password
-but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
-*/}}
-{{- if .Values.global -}}
-    {{- if .Values.global.storageClass -}}
-        {{- if (eq "-" .Values.global.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" .Values.global.storageClass -}}
-        {{- end -}}
-    {{- else -}}
-        {{- if $chartmuseum.storageClass -}}
-              {{- if (eq "-" $chartmuseum.storageClass) -}}
-                  {{- printf "storageClassName: \"\"" -}}
-              {{- else }}
-                  {{- printf "storageClassName: %s" $chartmuseum.storageClass -}}
-              {{- end -}}
-        {{- end -}}
-    {{- end -}}
-{{- else -}}
-    {{- if $chartmuseum.storageClass -}}
-        {{- if (eq "-" $chartmuseum.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" $chartmuseum.storageClass -}}
-        {{- end -}}
-    {{- end -}}
-{{- end -}}
+{{- include "common.storage.class" ( dict "persistence" .Values.persistence.persistentVolumeClaim.chartmuseum "global" $) -}}
 {{- end -}}
 
 {{/*
 Return the proper Storage Class for jobservice
 */}}
 {{- define "harbor.jobservice.storageClass" -}}
-{{- $jobservice := .Values.persistence.persistentVolumeClaim.jobservice -}}
-{{/*
-Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
-but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
-*/}}
-{{- if .Values.global -}}
-    {{- if .Values.global.storageClass -}}
-        {{- if (eq "-" .Values.global.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" .Values.global.storageClass -}}
-        {{- end -}}
-    {{- else -}}
-        {{- if $jobservice.storageClass -}}
-              {{- if (eq "-" $jobservice.storageClass) -}}
-                  {{- printf "storageClassName: \"\"" -}}
-              {{- else }}
-                  {{- printf "storageClassName: %s" $jobservice.storageClass -}}
-              {{- end -}}
-        {{- end -}}
-    {{- end -}}
-{{- else -}}
-    {{- if $jobservice.storageClass -}}
-        {{- if (eq "-" $jobservice.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" $jobservice.storageClass -}}
-        {{- end -}}
-    {{- end -}}
-{{- end -}}
+{{- include "common.storage.class" ( dict "persistence" .Values.persistence.persistentVolumeClaim.jobservice "global" $) -}}
 {{- end -}}
 
 {{/*
 Return the proper Storage Class for registry
 */}}
 {{- define "harbor.registry.storageClass" -}}
-{{- $registry := .Values.persistence.persistentVolumeClaim.registry -}}
+{{- include "common.storage.class" ( dict "persistence" .Values.persistence.persistentVolumeClaim.registry "global" $) -}}
```

Comment:
`"global"` should be `.Values.global`

Reply:
https://en.bloggif.com/tmp/2661767ce32fa2dea5c9383dadc372ea/text.gif?1589363719